test/shared_test/lib_battery_lifetime_test.cpp

```diff
@@ -0,0 +1,293 @@
+
+    while (idx < 1000){
+        if (idx % 2 != 0){
+            SOC = 90;
+        }
+        cal_model->runLifetimeCalendarModel(idx, T, SOC);
+        idx++;
+    }
+    s = cal_model->get_state();
+    EXPECT_NEAR(s.day_age_of_battery, 41, tol);
+    EXPECT_NEAR(s.q_relative_calendar, 101.69, tol);
+    EXPECT_NEAR(s.dq_relative_calendar_old, 0.00306, tol);
+}
+
+TEST_F(lib_battery_lifetime_calendar_model_test, replaceBatteryTest) {
+    double T = 4.85, SOC = 20;
+    int idx = 0;
+    while (idx < 200000){
+        if (idx % 2 != 0){
+            SOC = 90;
+        }
+        cal_model->runLifetimeCalendarModel(idx, T, SOC);
+        idx++;
+    }
+    calendar_state s = cal_model->get_state();
+    EXPECT_NEAR(s.day_age_of_battery, 8333, tol);
+    EXPECT_NEAR(s.q_relative_calendar, 97.67, tol);
+    EXPECT_NEAR(s.dq_relative_calendar_old, 0.043, tol);
+
+    cal_model->replaceBattery(5);
+
+    s = cal_model->get_state();
+    EXPECT_NEAR(s.day_age_of_battery, 0, tol);
+    EXPECT_NEAR(s.q_relative_calendar, 102, tol);
+    EXPECT_NEAR(s.dq_relative_calendar_old, 0.0, tol);
+}
+
+TEST_F(lib_battery_lifetime_calendar_matrix_test, TestLifetimeDegradation) {
+    double vals[] = { 0, 100, 365, 50 };
+    util::matrix_t<double> lifetime_matrix;
+    lifetime_matrix.assign(vals, 2, 2);
+
+    double dt_hour = 1;
+    lifetime_calendar_t hourly_lifetime(dt_hour, lifetime_matrix);
+
+    for (int idx = 0; idx < 8760; idx++) {
+        hourly_lifetime.runLifetimeCalendarModel(idx, 20, 80);
+    }
+
+    EXPECT_NEAR(hourly_lifetime.capacity_percent(), 50, 1);
+
+    dt_hour = 1.0 / 12.0; // Every 5 mins
+    lifetime_calendar_t subhourly_lifetime(dt_hour, lifetime_matrix);
+
+    for (int idx = 0; idx < 8760 * 12; idx++) {
+        subhourly_lifetime.runLifetimeCalendarModel(idx, 20, 80);
+    }
+
+    EXPECT_NEAR(subhourly_lifetime.capacity_percent(), 50, 1);
+}
+
+
+TEST_F(lib_battery_lifetime_calendar_model_test, TestLifetimeDegradation) {
+
+    for (int idx = 0; idx < 8760; idx++) {
+        cal_model->runLifetimeCalendarModel(idx, 20, 80);
+    }
+
+    EXPECT_NEAR(cal_model->capacity_percent(), 99.812, 1);
+
+    dt_hour = 1.0 / 12.0; // Every 5 mins
+    lifetime_calendar_t subhourly_lifetime(dt_hour);
+
+    for (int idx = 0; idx < 8760 * 12; idx++) {
+        subhourly_lifetime.runLifetimeCalendarModel(idx, 20, 80);
+    }
+
+    EXPECT_NEAR(subhourly_lifetime.capacity_percent(), 99.812, 1);
+}
+
+TEST_F(lib_battery_lifetime_test, updateCapacityTest) {
+    size_t idx = 0;
+    while (idx < 876){
+        model->runLifetimeModels(idx, true, 5,95, 25);
+        model->runLifetimeModels(idx, true, 95, 5, 25);
+
+        auto state = model->get_state();
+        EXPECT_EQ(state.cycle->q_relative_cycle, model->capacity_percent_cycle());
+        EXPECT_EQ(state.calendar->q_relative_calendar, model->capacity_percent_calendar());
+
+        idx ++;
+    }
+
+}
+
+
+//TEST_F(lib_battery_lifetime_test, ReplaceByCapacityTest){
```

Comment:
Can you add a comment with what needs to happen to get this test uncommented?

Reply:
I had moved the replacements out of lifetime and into the battery, so I cleaned up the test and moved it as part of lib_battery_test